libcudacxx/.upstream-tests/test/std/iterators/iterator.requirements/indirectcallable/indirectinvocable/indirect_strict_weak_order.compile.pass.cpp

```diff
@@ -48,6 +48,10 @@ struct GoodOrder {
 // Should work when all constraints are satisfied
 static_assert(cuda::std::indirect_strict_weak_order<GoodOrder<It1, It2>, It1, It2>);
 static_assert(cuda::std::indirect_strict_weak_order<bool(*)(int, long), int*, long*>);
+
+#ifdef TEST_COMPILER_CLANG_CUDA
+#pragma clang diagnostic ignored "-Wunneeded-internal-declaration"
+#endif // TEST_COMPILER_CLANG_CUDA
 #ifndef __CUDA_ARCH__
 auto lambda = [](int i, long j) { return i == j; };
 static_assert(cuda::std::indirect_strict_weak_order<decltype(lambda), int*, long*>);
```

Comment:
@Artem-B this one might be interesting for you.

the lambda is only ever used within the `static_assert` which triggers an "unneeded-internal-declaration" warning.

Reply:
I think clang's warning may be valid. AFAICT, the variable is unused. Static assert only uses its type.

You could avoid that by using something like this (or just by inlining the lambda into the static assert.:
```
using lambda_t = decltype([](int i, long j) { return i == j; });
static_assert(cuda::std::indirect_strict_weak_order<lambda_t, int*, long*>);
```